webgui/scripts/codecompass/urlHandler.js

```diff
@@ -15,15 +40,17 @@ function (lang, hash, topic) {
      */
     updateUrl : function () {
       var that = this;
-      
+
       var hashValue = [];
 
       for (var s in state)
         hashValue.push(s + '=' + state[s]);
       this.hashSetProgress = true;
-      
+
       hash(hashValue.join('&'));
-      
+
+      this._cacheDbInformation();
```

Comment:
This function call will make many unnecessary queries for file content, file info and AST node info. These are relatively large objects and will always be queried even if URL change isn't related of file or node change at all.